### ISSUE TITLE
node 0.10.28 causes several issues 

### ISSUE THREAD

**Issue by fine129** [cgiffard/node-simplecrawler#79]
I used node 0.8.26 and the crawler worked fine.
but several days ago i  upgraded node to 0.10.28,
several problems occurred.

1\ In handleResponse function , every if clause seems to need: response.on("data",function(){});
 to fire the "end" event of the response. Otherwise the process will freeze .
2\In handleResponse function, memory leak( process out of memory error) occurred   after crawling several hours.It is very strange.I didn't see this kind of problems while using node 0.8.26.
3\In https mode,using node 0.10.28,I encountered "self_signed_cert_in_chain" error.
I googled this answer: http://loutilities.wordpress.com/2013/04/08/using-ssl-self-signed-certs-with-node-js/

I spent a whole day trying to solve these issues.Sadly i got no success .

**Comment by pinacodelada** [comment]
If I am seeing rogue "fetchtimeout"s with any requests that are not a 200 or 304 http status code, am I in the right place?

**Comment by cgiffard** [comment]
@pinacodelada Please create a new issue, if possible. I haven't been able to reproduce any of the issues in this bug without a test case, so I'm going to close it for now.

If you could supply a test case for your issue, it would be much appreciated!

**Comment by fantapop** [comment]
So, I think that I ran into this.   The description is very similar. Using node 0.10.35,  I'm crawling a site and the progress stops at a certain point.  That point changes if I change the number of allowed concurrent requests.  I simplified settings and added a ton of logging to dig into this.  I never fully figured out what's wrong but the queue effectively freezes because there are some requests that are fired off through the https.request() method and never come back.  As mentioned above, this is very consistent.  I added handlers to all known events fired by the client request and none of them are firing.  I upgraded to node 4.1.1 and whatever the issue was, it seems to be fixed.  Upgrading node is fine for me for this use case so I'm not going to debug this any further.  I just wanted to post my findings here in case they help anyone else.
